pkg/api/controller.go

```diff
@@ -96,8 +99,23 @@ func (c *Controller) Run() error {
 			}
 		}
 
-		defaultStore := storage.NewImageStore(c.Config.Storage.RootDirectory,
-			c.Config.Storage.GC, c.Config.Storage.Dedupe, c.Log)
+		var defaultStore storage.ImageStore
+		if len(c.Config.Storage.StorageDriver) == 0 {
+			defaultStore = storage.NewImageStore(c.Config.Storage.RootDirectory,
+				c.Config.Storage.GC, c.Config.Storage.Dedupe, c.Log)
+		} else {
+			storeName := fmt.Sprintf("%v", c.Config.Storage.StorageDriver["name"])
```

Comment:
Check and enforce it is "s3" driver only until we add support for others.

Reply:
Fixed

Reply:
One more comment. Move this "s3" into a pkg/storage/storage.go as some constant so we don't compare strings directly. Something like this.


```
const (
S3StorageDriverName = "s3"
)
```